tutorial_devops/consuming_conan_center.rst

```diff
@@ -0,0 +1,52 @@
+- Fixes to ConanCenter recipes can take a while to be merged (weeks), because there are too many contributions to be reviewed. If you rely on a fix that you contributed and it is urgent for your production system, you might need to wait.
+- Binary configurations can be discontinued in ConanCenter, when new compiler versions are released and ConanCenter starts to build them, it can drop some of the older versions.
+- Outages, maintenance windows and other service interruptions can happen, ranging from a few minutes to days. The infrastructure to host central repositories is typically large and complex, and relies on cloud providers that sometimes have their own issues, networks, CDN, etc.
+- Security issues. Even if the process to add packages to ConanCenter is very strict, with human reviews for every package and many automated checks and processes, eventually all systems are subject to attempts of supply chain attacks.
+
+Most of these reasons are not unique to Conan, but to every other package manager for other programming languages. The general devops known good practice is that you shouldn't rely on central repositories on production systems, but host your own copy of the packages that you need for production.
+
+This is the main idea: **host your own copy of the packages you need for production in your own server**. This could be done by just downloading binaries from ConanCenter and then uploading them to your own server. However, it is much better to fully own the complete supply chain and create the binaries in your own CI systems. So the recommended flow to use ConanCenter packages in production would be:
+
+- Create a fork of the ConanCenter Github repository: https://github.com/conan-io/conan-center-index
+- Create a list of the packages and versions you need for your projects. This list can be added to the fork too, and maintained there (packages can be added and removed with PRs when the teams need them).
+- Create a script that first ``conan export`` all the packages in your list, then ``conan create --build=missing`` them. Do not add ``user/channel`` to these packages, it is way simpler to use them as ``zlib/1.2.13`` without user-channel.
+- Upload your build packages to your own server, that you use in production, instead of ConanCenter.
+
+
+This flow is relatively straightforward, and has many advantages that mitigate the above risks:
+
+- No central repository outage can affect your builds.
+- No changes in the central repository can break your projects.
+- You can customize, adapt, fix and perfectly control what versions are used, and release fixes in minutes, not weeks. You can apply customizations that wouldn't be accepted in the central repository.
+- You fully own the supply chain, from the source (recipes) to the binaries, eliminating in practice the majority of potential supply chain attacks of central repositories.
```

Comment:
Arguable too - for recipes yes, but any large project that uses "a lot" of open source recipes, they are still going to be pulling sources (of the libraries) from a variety of places - those are rarely ever reviewed (does anyone look at the sources of boost??) - and at most the validation is checksum checking. Each of these many repositories / source packages hosted on a plethora of platforms _is_ liable to supply chain attacks. As a reader, to me "fully owning" the supply chain means having forks of all those libraries. I think hosting your own COnan Center puts you closer, but it's not full control.

Reply:
A bit of rewording, but I think the original claims are correct:
- You full control the binaries supply chain: your compilers, your tools, your computers. You fully control the sources that are used
- The attacks that are eliminated are those of "central repositories" not all attacks of supply chains. If someone hacks the boost sources, you are still very exposed, but not as a result of ConanCenter being attacked, and the same if you didn't use Conan. The important bit is that Conan itself does not introduce new attack vectors